hoodie-common/src/main/java/com/uber/hoodie/configs/AbstractJobConfig.java

```diff
@@ -0,0 +1,151 @@
+/*
+ *  Copyright (c) 2017 Uber Technologies, Inc. (hoodie-dev-group@uber.com)
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *           http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ */
+
+package com.uber.hoodie.configs;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+import com.uber.hoodie.exception.InvalidJobConfigException;
+import java.io.Serializable;
+import java.lang.reflect.Field;
+import java.util.Map;
+
+public abstract class AbstractJobConfig implements Serializable {
+
+  private static final String SPACE = " ";
+
+  @Parameter(names = {"--help", "-h"}, help = true)
+  public Boolean help = false;
+
+  public String[] getJobConfigsAsCommandOption(String prefix) {
+    return getJobConfigsAsCommandOption(prefix, null);
+  }
+
+  public String[] getJobConfigsAsCommandOption(String prefix, String suffix) {
+    StringBuilder result = new StringBuilder();
+    Field[] fields = this.getClass().getDeclaredFields();
```

Comment:
Wondering if JCommander has any such API to have similar functionality

Reply:
Unfortunately, haven't able to find any such utility functionality in Jcommander package.